component/http/v2/component.go

```diff
@@ -0,0 +1,95 @@
+// Package v2 provides a ready to use HTTP component.
+package v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/beatlabs/patron/log"
+)
+
+const (
+	port                = 50000
+	readTimeout         = 30 * time.Second
+	writeTimeout        = 60 * time.Second
+	idleTimeout         = 240 * time.Second
+	shutdownGracePeriod = 5 * time.Second
+)
+
+// Component implementation of an HTTP router.
+type Component struct {
+	port                int
+	readTimeout         time.Duration
+	writeTimeout        time.Duration
+	shutdownGracePeriod time.Duration
+	handler             http.Handler
+	sync.Mutex
```

Comment:
It probably shouldn't be an embedded type as it promotes the `Lock`/`Unlock` method.

Reply:
done